### PR TITLE
KAFKA-17814 Use `final` declaration to replace the suppression `this-escape`

### PR DESCRIPTION
**Pull request by Linsiyuan9**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by frankvicky** [comment]
Hello @Linsiyuan9 
Thanks for the patch.
Could you please revise the title and description ?

**Comment by Linsiyuan9** [comment]
Hi @frankvicky ,Sorry for replying you so late. I have modified it

File: clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslClientProvider.java [hunk 1]
```diff
@@ -25,14 +25,14 @@
 public class OAuthBearerSaslClientProvider extends Provider {
     private static final long serialVersionUID = 1L;
 
-    @SuppressWarnings("this-escape")
-    protected OAuthBearerSaslClientProvider() {
+    private OAuthBearerSaslClientProvider() {
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslClientProvider.java:28]: Could you please use `final` to fix it?

File: clients/src/main/java/org/apache/kafka/common/security/plain/internals/PlainSaslServerProvider.java [hunk 2]
```diff
@@ -25,13 +25,13 @@ public class PlainSaslServerProvider extends Provider {
 
     private static final long serialVersionUID = 1L;
 
-    @SuppressWarnings("this-escape")
-    protected PlainSaslServerProvider() {
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/security/plain/internals/PlainSaslServerProvider.java:29]: ditto

File: clients/src/main/java/org/apache/kafka/common/security/scram/internals/ScramSaslClientProvider.java [hunk 3]
```diff
@@ -25,14 +25,14 @@ public class ScramSaslClientProvider extends Provider {
 
     private static final long serialVersionUID = 1L;
 
-    @SuppressWarnings("this-escape")
-    protected ScramSaslClientProvider() {
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/security/scram/internals/ScramSaslClientProvider.java:29]: ditto

File: clients/src/main/java/org/apache/kafka/common/security/scram/internals/ScramSaslServerProvider.java [hunk 4]
```diff
@@ -25,14 +25,14 @@ public class ScramSaslServerProvider extends Provider {
 
     private static final long serialVersionUID = 1L;
 
-    @SuppressWarnings("this-escape")
-    protected ScramSaslServerProvider() {
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/security/scram/internals/ScramSaslServerProvider.java:29]: ditto

**Review by chia7712 — COMMENTED** [review]
@Linsiyuan9 thanks for this refactor

**Review by chia7712 — APPROVED** [review]

**Pull request merged**